### ISSUE TITLE
MoreKnockback

### ISSUE THREAD

**Issue by Dbbtop** [GrimAnticheat/Grim#1780]
### Describe the bypass and how to replicate it

```java
@EventHandler
    public void onSendPacket(PacketEvent.Send event) {
        if ((!MovementUtility.isMoving() || inMove.getValue())
                && event.getPacket() instanceof PlayerInteractEntityC2SPacket
                && getInteractType(event.getPacket()) == Criticals.InteractType.ATTACK
                && !(getEntity(event.getPacket()) instanceof EndCrystalEntity)
                && getEntity(event.getPacket()) instanceof LivingEntity lent
                && lent.hurtTime <= hurtTime.getValue()
                && MathUtility.random(0, 100) >= (100 - chance.getValue())
                && !canCrit()) {

            if (mc.player.isSprinting()) sendPacket(new ClientCommandC2SPacket(mc.player, ClientCommandC2SPacket.Mode.STOP_SPRINTING));
            sendPacket(new ClientCommandC2SPacket(mc.player, ClientCommandC2SPacket.Mode.START_SPRINTING));
            sendPacket(new ClientCommandC2SPacket(mc.player, ClientCommandC2SPacket.Mode.STOP_SPRINTING));
            sendPacket(new ClientCommandC2SPacket(mc.player, ClientCommandC2SPacket.Mode.START_SPRINTING));
            debug("wtap");
            mc.player.setSprinting(true);
            mc.player.lastSprinting = true;
```

### Grim version

latest

### Server version

1.19.4

### Plugins

grim, packetevents, LP

**Comment by ManInMyVan** [comment]
patched by #1678

**Comment by Dbbtop** [comment]
> patched by #1678

i just tested it, and nah

**Comment by ManInMyVan** [comment]
> > patched by #1678
> 
> i just tested it, and nah

did you put "PacketOrder" in your punishments.yml?

**Comment by Dbbtop** [comment]
> > > patched by #1678
> > 
> > 
> > i just tested it, and nah
> 
> did you put "PacketOrder" in your punishments.yml?

Yes

**Comment by ManInMyVan** [comment]
> > > > patched by #1678
> > > 
> > > 
> > > i just tested it, and nah
> > 
> > did you put "PacketOrder" in your punishments.yml?
> 
> Yes

did you enable experimental checks?

**Comment by Dbbtop** [comment]
Yes

**Comment by ManInMyVan** [comment]
did you enable verbose?

**Comment by Dbbtop** [comment]
Yes

**Comment by ManInMyVan** [comment]
> Yes

try using this jar https://github.com/ManInMyVan/Grim/actions/runs/11761716232/artifacts/2167594681

**Comment by Dbbtop** [comment]
> > Yes
> 
> try using this jar https://github.com/ManInMyVan/Grim/actions/runs/11761716232/artifacts/2167594681

 tested it

**Comment by Dbbtop** [comment]
> > Yes
> 
> try using this jar https://github.com/ManInMyVan/Grim/actions/runs/11761716232/artifacts/2167594681

https://mega.nz/file/WyYjyK7Y#Ua1cXHyf52YSRRI4TgrolVxrlR4LXfRW1abX0kK0JuI

**Comment by SuicidalEeper** [comment]
The code you provided above is only supposed to actually run when you're sprinting though no?

**Comment by Dbbtop** [comment]
this function sends a sprint packet to make a pushback attack even without movement

**Comment by ManInMyVan** [comment]
> this function sends a sprint packet to make a pushback attack even without movement

it doesn't look like you have that enabled

**Comment by ManInMyVan** [comment]
> this function sends a sprint packet to make a pushback attack even without movement

I'm not checking stationary players on 1.9+ because of tick skipping

**Comment by Axionize** [comment]
As far as I can tell this is literally just an auto w-tap that only works when you're standing still. When would this be useful as hack?

This would only be useful if you can use it while moving...

Please correct me if I'm wrong or misunderstanding

**Comment by Dbbtop** [comment]
No, this is not an auto hit. This hack sends 2 packets when hitting - start_sprint and stop_sprint, thus, the target entity is pushed back more than with a normal hit without the hack. This works even when the hacker is not moving. This is an abuse of the mechanics of Minecraft

**Comment by Axionize** [comment]
That is not what I'm saying. As the debug statement in the code you copy pasted literally says, all this is a packet w-tap cheat. 

It's detected by BadPacketsF as soon as this is used during an attack while a player resprints (and they're moving), which happens every so often in actual fights.

If you don't want to get detected (and therefore simply hold w in a fight with this cheat module on) against a player who actually knows what they're doing you will lose.

The point I was trying to make is because of the reasons I have explained above this is a very low priority "bypass." This can definitely be *better* detected through heuristics but obviously Grim 2.0 has none of them unlike Grim Premium/3.0 because the moment you make such a check open source it becomes worthless.

We may however be able to design a better logic check for this.

**Comment by Axionize** [comment]
@SamB440 can we copy your private check for this into Grim or is that considered a heuristic?

**Comment by MitroxYT** [comment]
This may not be the right approach but 
many cheats can be fixed
by checking for a gap between packages and 
this will fix many cheats like
Autotem
Autoswap
Chest stealer
Translated by DeepL
.

**Comment by Axionize** [comment]
"Gap between packages?"
???

**Comment by ManInMyVan** [comment]
> "Gap between packages?"
> ???

delay between packets

**Comment by Axionize** [comment]
@SamB440 Can I please copy your Pharus check into here to just completely fix this for now?

**Comment by Axionize** [comment]
@SamB440 you said you'd port a new version of the Pharus check, how's that going?
